code/game/objects/items/weapons/tools/_tools.dm

```diff
@@ -898,6 +904,16 @@
 			ratio = max(round(ratio, 0.25) * 100, 25)
 			overlays += "[icon_state]-[ratio]"
 
+	if(ismob(loc))
+		var/tooloverlay
+		switch(mode)
+			if (EXCAVATE)
+				tooloverlay = "excavate"
+			if (DIG)
+				tooloverlay = "dig"
+		overlays += (tooloverlay)
+		update_wear_icon()
```

Comment:
That line can be ommited `update_wear_icon()` will be called by `turn_on`/`turn_off` code anyway.

Reply:
Requested change made on local repository.